src/Microsoft.Health.Fhir.SqlServer/Features/ChangeFeed/SqlServerFhirResourceChangeDataStore.cs

```diff
@@ -0,0 +1,78 @@
+{
+    public class SqlServerFhirResourceChangeDataStore : IChangeFeedSource<IResourceChangeData>
+    {
+        private readonly ISqlConnectionFactory _sqlConnectionFactory;
+        private readonly ILogger<SqlServerFhirResourceChangeDataStore> _logger;
+
+        public SqlServerFhirResourceChangeDataStore(ISqlConnectionFactory sqlConnectionFactory, ILogger<SqlServerFhirResourceChangeDataStore> logger)
+        {
+            EnsureArg.IsNotNull(sqlConnectionFactory, nameof(sqlConnectionFactory));
+            EnsureArg.IsNotNull(logger, nameof(logger));
+
+            _sqlConnectionFactory = sqlConnectionFactory;
+            _logger = logger;
+        }
+
+        public async Task<IReadOnlyCollection<IResourceChangeData>> GetRecordsAsync(long startId, int pageSize, CancellationToken cancellationToken)
+        {
+            EnsureArg.IsGte(startId, 0, nameof(startId));
+            EnsureArg.IsGte(pageSize, 0, nameof(pageSize));
+
+            var listResourceChangeData = new List<ResourceChangeData>();
+            try
+            {
+                using (SqlConnection sqlConnection = await _sqlConnectionFactory.GetSqlConnectionAsync(cancellationToken: cancellationToken))
+                {
+                    await sqlConnection.OpenAsync(cancellationToken);
+                    using (SqlCommand sqlCommand = new SqlCommand("dbo.FetchResourceChanges", sqlConnection))
+                    {
+                        sqlCommand.CommandType = CommandType.StoredProcedure;
+                        sqlCommand.Parameters.AddWithValue("@startId", SqlDbType.BigInt).Value = startId;
+                        sqlCommand.Parameters.AddWithValue("@pageSize", SqlDbType.Int).Value = pageSize;
+
+                        using (SqlDataReader sqlDataReader = await sqlCommand.ExecuteReaderAsync(CommandBehavior.SequentialAccess, cancellationToken))
+                        {
+                            while (await sqlDataReader.ReadAsync(cancellationToken))
+                            {
+                                listResourceChangeData.Add(new ResourceChangeData
+                                {
+                                    Id = (long)sqlDataReader["Id"],
+                                    Timestamp = DateTime.SpecifyKind((DateTime)sqlDataReader["Timestamp"], DateTimeKind.Utc),
+                                    ResourceId = (string)sqlDataReader["ResourceId"],
+                                    ResourceTypeId = (short)sqlDataReader["ResourceTypeId"],
+                                    ResourceVersion = (int)sqlDataReader["ResourceVersion"],
+                                    ResourceChangeTypeId = (byte)sqlDataReader["ResourceChangeTypeId"],
+                                });
+                            }
+                        }
+
+                        return listResourceChangeData;
+                    }
+                }
+            }
+            catch (Exception ex)
```

Comment:
Questions: are there any different types of exceptions we can pivot here? For example, we can differentiate these scenarios:

1. Intermittent connection error => we can retry
2. Un-processable data => we should not retry, as there could be a bug.
3. Other unknown exception => just log them.

Reply:
Are you saying that "retry logic" should be in this function? or we need to define custom exception classes. I think "retry logic", and handling exceptions should be on the caller side (publish event service).

Reply:
One thing I would do is use this to document the potential exception types that could be thrown ( like SqlException, TimeoutException etc) https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/xmldoc/exception

Reply:
Added comments and exceptions.